bser/decode_test.go

```diff
@@ -1 +1,96 @@
 package bser
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+type decodeTest struct {
+	encoded      []byte
+	expectedData interface{}
+	expectErr    bool
+	doDecode     func(decoder *Decoder) (interface{}, error)
+}
+
+var decodeTests = map[string]decodeTest{
+	"single_object": {
+		encoded: []byte(
+			"\x00\x01\x05\x19\x00\x00\x00\x01\x03\x02\x02\x03\x04Name\x02\x03\x04fred\x02\x03\x03Age\x03\x14",
+		),
+		expectedData: person{Name: "fred", Age: 20},
+		doDecode: func(decoder *Decoder) (interface{}, error) {
+			dst := person{}
+			err := decoder.Decode(&dst)
+			return dst, err
+		},
+	},
+}
+
+func TestDecode(t *testing.T) {
+	for testName, testCase := range decodeTests {
+		t.Run(testName, func(t *testing.T) {
+			testDecode(t, testCase)
+		})
+	}
+}
+
+func testDecode(t *testing.T, testCase decodeTest) {
+	var (
+		buf     = bytes.NewBuffer(testCase.encoded)
+		decoder = NewDecoder(buf)
+	)
+	t.Helper()
```

Comment:
can you make this the first line in the func?

Reply:
done